Smashing Magazine heat map

Davidosky007 opened the pull request:
in this project we :
* used flex and grid for positioning of contents on the page
* used google fonts for fonts styling
* also applied the necessary semantic elements where necessary

Davidosky007 committed:
add nav section

henatan99 committed:
Latest articles section added

- Latest articles section elements are added
- Four article sections are layedout in a grid
- Background colors are added for grid elements
- Gradient background is added to the articles header and aligned the
  colors with the grid elements

Davidosky007 committed:
add animation

Davidosky007 committed:
update README file

Davidosky007 committed:
add membership section

Davidosky007 committed:
Merge branch 'feature-1' of github.com:Davidosky007/Smashing-Magazine into feature-1

henatan99 committed:
Add Latest Posts, Smashing Membership and New on Smashing TV

- Align Latest posts elements in a flex
- Apply Traslate and transition effects
- Apply other hover effects

henatan99 committed:
Communtiy links and footer sections added and styled

- Commmunity links has one person of the week div section. And another
  div elemet wrapping other child elements in grid layout.
- The footer section has three child elemets aligned in flex column
  layout. The list section elements are aligned in flex-flow row wrap
  layout.

Davidosky007 committed:
add ship now section

henatan99 committed:
Add our guide section, more style footer section

- Add elements of our guide section using flex-flow row wrap layout
- Modify styles to footer section and add remaining sections

Davidosky007 committed:
fix style sheet error

Davidosky007 committed:
modify html file

henatan99 committed:
Change elements name to more semantic and add on footer

- Change to more semantic names to elements
- Add the images and the fixed arrow button in the footer section

Davidosky007 committed:
add design system section and modify README file

henatan99 committed:
Changes more semantic names for elements, google fonts

- Add more sematic names like article
- Add few google fonts
- Add more final styling

Davidosky007 committed:
fix linter errors and modify style sheet

Davidosky007 committed:
style sheet and linter error modification

jubaan reviewed (changes requested):
# Status: Changes Required :recycle:

## Amazing project team!! 
:100: love all the effects in the site
:100: It's pretty close to the original (only missing the color)
:100: great implementation of the heatmap
:100: Well defined sections in CSS file
:100: Great README
;100: overall a great looking project

## Required Changes
### [Mandatory] :red_circle: 

- [ ] Logo should be part of the navbar

- [ ] Search should be part of the navbar

- [ ] Fix indentation, not all elements are aligned correctly through the HTML file. A well-formatted code helps readability. See inline comments.

- [ ] Change footer menus to ```<ul>``` with ```<li>```. This is more semantic than the ```<div>```. See inline comments

- [ ] wrap the articles sections in a ```<main>``` HTML tag.

- [ ] Leave out the cursor pointer and convert link elements to real links with and anchor ```<a>``` tag. While it works by mimicking the website functionality is better and easier to use and anchor tag, also is more semantic.

### [OPTIONAL]  :large_blue_circle: 

- [ ] Maintain code DRY. See inline comments

## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

jubaan commented on index.html:
```diff
@@ -0,0 +1,846 @@
+  </header>
+
+
+  <!-- LATEST ARTICLES SECTION -->
+
+  <section class="latest-articles">
+    <div class="notify-article">
+      <h1 class="first-notify">
+        Don't Miss These Latest
+      </h1>
+      <h1 class="second-notify">
```
wrap all articles section in a ```<main>``` container

jubaan commented on index.html:
```diff
@@ -0,0 +1,846 @@
+          Books
+        </a>
+        </li>
+        <li>
+        <a class="nav-links" href="#">
+            Events
+        </a>
+        </li>
```
Fix the indentation of all elements that require it.

jubaan commented on index.html:
```diff
@@ -0,0 +1,846 @@
+          Membership
+        </a>
+        </li>
+        <li>
+          <a class="btn-nav-links" href="#">
+              <span class="btn-text">
+                More
+              </span>
+              <img src="./images/menu-cat.svg" alt="menu-cat">
+        </a>
```
Correct indentation, I just pointed out some, but there are more indentation issues.

jubaan commented on index.html:
```diff
@@ -0,0 +1,846 @@
+    </h2>
+    <div class="features-link">
+      <a href="#" class="explore">Explore Features</a>
+      <span>&#8594;</span> 
```
The <span> tag is an inline container used to mark up a part of a text, or a part of a document. The correct use of span is as follows:
```
<p>My mother has <span style="color:blue">blue</span> eyes.</p>
```

jubaan commented on index.html:
```diff
@@ -0,0 +1,846 @@
+    <h3><small>BROWSE ALL SMASHING MAGAZINE TOPICS</small></h3>
+  </div>
+
+  <div class="footer-list">
+    <div>
+      <img src="images/bullet.svg" alt="bullet">
+      <a href="#">Android</a>      
+    </div>    
+    <div>
+      <img src="images/bullet.svg" alt="bullet">
+      <a href="#">Animation</a> 
+    </div>
+    <div>
+      <img src="images/bullet.svg" alt="bullet">
+      <a href="#">Apps</a> 
+    </div>
+    <div>
+      <img src="images/bullet.svg" alt="bullet">
+      <a href="#">CSS</a> 
+    </div>
+    <div>
+      <img src="images/bullet.svg" alt="bullet">
+      <a href="#">Deisgn</a> 
+    </div>
+    <div>
+      <img src="images/bullet.svg" alt="bullet">
+      <a href="#">Design Patterns</a> 
+    </div>
+    <div>
```
A better approach is using ```<ul>``` and ```<li>```

jubaan commented on index.html:
```diff
@@ -0,0 +1,846 @@
+    <span>Fonts by <a href="#">Latinotype</a>.</span>
+  </div>
+
+  <div class="footer-links">
+    <a href="#">✎ Write for us</a>
+    <a href="#">Contact us</a>
+    <a href="#">About us (Impressum)</a>
+    <a href="#">Privacy policy</a>
+    <a href="#">Membership login</a>
+    <a href="#">Delivery times</a>
+    <a href="#">Advertise</a>    
+  </div>  
```
The same applies for this. ```<ul>``` and ```<li>```

jubaan commented on index.html:
```diff
@@ -0,0 +1,846 @@
+    </div>
+  </div>
+
+  <div class="statement">
+    <span>With a commitment to quality content for the design community.</span>
+    <span>Founded by Vitaly Friedman and Sven Lennartz. 2006–2020.</span>
+    <span>Smashing is proudly running on <a href="#">Netlify</a>.</span>
+    <span>Fonts by <a href="#">Latinotype</a>.</span>
+  </div>
```
The same comment as above with the use of ```<span>```

jubaan commented on index.html:
```diff
@@ -0,0 +1,846 @@
+  <span class="iconify" data-icon="dashicons:arrow-up-alt2" data-inline="false"></span>
+</a>  
+
+
+</body>
+
+</html>
```
Indentation

jubaan commented on style.css:
```diff
@@ -0,0 +1,1043 @@
+  color: rgb(59, 53, 53);
+}
+
+.post-intro h2 {
+  margin-top: 20px;
+  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
+  font-weight: 100;
+  line-height: 32px;
+}
+
+.post-author {
+  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
+  font-size: 15px;
+  font-weight: 800;
+}
```
to make the code DRYer a good approach is to take out repeated properties and create their own selectors. In this case with the ```font-family```, my perception is that is used through all the file as the default font.

To improve it this could be done:

```
/* -- At the top of the file -- */
*,
* a {
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}
```
or if the case is that only links and titles and subtitles are the ones that require this anther approach would be:
```
h1,
h2,
h3,
* a,
.post-intro,
.post-author {
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}
```

henatan99 committed:
Changes made after reviewer comments

- Improper use of span tag changed
- List elements are wrapped with ul and li tags instead of other
  non-semantic tags
- Apply code drying
- Apply document formatting

Davidosky007 committed:
made changes to the nav section and fix some linter errors

umairarshadbutt reviewed (approved):
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
### The [clone](https://davidosky007.github.io/Smashing-Magazine/) shows a good understanding of Design Teardown. :100:
##
- The cloned page is rendering as it should and meets all the project specifications.
- The project has a descriptive README.md and is professionally done. Good job with that.
- Pull Request has a meaningful title and a short description.
- CSS is clean. No linter errors at this point and no bad practices used as well.
- The repo has followed all the guidelines there are to follow (GitHub flow etc.)
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.**
- Congratulations on the splendid job @henatan99 and @davidosky007! :+1: 
- You can now merge to master and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal:
##
**If you have any questions, let me know on slack at any time. I will be happy to help.**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)
